spec/definitions.yaml

```diff
@@ -1034,6 +1052,20 @@ definitions:
             type: string
             description: |
               Azure virtual machine size used by all nodes in this pool.
+          spot_instances:
+            description: |
+              Attributes defining the presence and configuration of Spot Instances.
```

Comment:
I would expect a bit more information about what spot instances are and what this really means for the node pool. Ideally we would link out to a good Azure docs page for the first.

Reply:
Shouldn't that information be in the docs, rather than the API spec?

Reply:
Yeah, also fine. Then a link to these docs would be great (once they exist).